### PR TITLE
[492072] Minimized usage of local variables

### PR DESCRIPTION
**Pull request by kthoms**
The Xtend compiler was optimized to minimize the usage of synthetic variables for member feature calls and right-hand assignments

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=492072

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/FeatureCallCompiler.java [hunk 1]
```diff
@@ -504,10 +504,9 @@ protected void prepareExpression(XExpression arg, ITreeAppendable b) {
 			XAbstractFeatureCall featureCall = (XAbstractFeatureCall) arg;
 			// check for final modifier
 			if (featureCall.getFeature() instanceof XVariableDeclaration) {
-				XVariableDeclaration variableDeclaration = (XVariableDeclaration) featureCall.getFeature();
+				XVariableDeclaration variableDeclaration = (XVariableDeclaration) featureCall.getFeature();;
```
Comment (spoenemann) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/FeatureCallCompiler.java]: double semicolon?

Comment (kthoms) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/FeatureCallCompiler.java]: Will fix this ASAP

Comment (kthoms) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/FeatureCallCompiler.java]: Done

**Comment by kthoms** [comment]
Build is green: http://services.typefox.io/open-source/jenkins/job/xtext-extras/job/kth_bug492072/

**Comment by kthoms** [comment]
@spoenemann OK to merge?

**Comment by holgerschill** [comment]
Looks good to me, but I would prefer to have someone else say +1.

**Comment by kthoms** [comment]
@spoenemann This fix is urgent, the current Xtend SDK is broken without it. Please also look at eclipse/xtext-xtend#106 which adds tests and reverts a workaround I tried, but it was the wrong place to fix it effectively.

**Comment by spoenemann** [comment]
I don't know enough about that code to really understand the consequences of this change.

**Pull request merged**